[KIALI-343] Traffic Generator to Kiali Test Mesh

gbaufake opened the pull request:
Similarly to watch -n 1 curl -o /dev/null -s -w %{http_code}\n productpage-istio-system.route, this docker-image generates to loads in order to have some data on service-graph.

@mwringe
I think this traffic generator may help to validate the traffic between the different service mesh types.

mwringe commented on traffic-generator/openshift/traffic-generator.yml:
```diff
@@ -0,0 +1,55 @@
+
+  - name: ROUTE
+    description: What is the ROUTE to be queried
+    value: CHANGEME
```
We should remove the 'value' here and make the parameter required. This will force someone to set a value

mwringe commented:
A couple of comments:

- What Jira is this targeting? We should always have a Jira created for a task where it can be discussed as to how we want to accomplish something. The commit message should also start with the Jira name.

- can you please move the existing service code to a directory called something like 'test-service', and then have this under a directory called 'traffic-generator'. I think this will make things easier to understand.

- There is no readme or anything to describe how to use this. Can we include some example type deployments like how the existing hack script does things?

- A single path is not enough to generate some of the different types of service meshes. We would need multiple paths for that.

- our code is in go lang and typescript already, do we really need to introduce another language?

gbaufake commented:
I updated the traffic generator with 
- A golang load generator (in order to use the same code language)
- Moved to test service directory
- A small README indicating how to use it (a parallel script to deploy on openshift can be created as well)
- Openshift Template with parameters

mwringe commented on test-service/README.adoc:
```diff
@@ -0,0 +1,46 @@
+
+= Introduction
+
+Kiali Traffic Generator is a component which generate then ecessary traffic for visualization of Kiali Test Meshes on Kiali Service Graph.
```
type: "then ecessary"

mwringe commented:
I think what we want at the root of this is:

README.adoc (updated to include the load generator)
LICENSE
test-service (everything else we had before the commit)
load-generator (the new things you introduced)

mwringe commented on test-service/README.adoc:
```diff
@@ -0,0 +1,46 @@
+
+- *You may use mutiple routes. They must be separate by ; (semicolon)*
+`eg: http://a-kiali-test-breath.your.openshift.com/route?path=a;http://a-kiali-test-breath.your.openshift.com/route?path=b`
+
```
The urls here are wrong, instead of 'a-kiali-test-breath.your.openshift.com' just use 'a', the dns system used by OpenShift/k8s will resolve this to a service which is running in the same project.

mwringe commented:
I think we also need to move the configuration of this to a configmap

mwringe commented:
When I mentioned it might be good to use a golang based application instead of python, I meant writing something to do the load generation using golang, I didn't mean to use bash to call an application that just happens to be written in go :)

But for now that is fine.

mwringe commented:
Overall, what I think we want here is a way to deploy to an OpenShift project a bunch of test-services and the load generator using a bash script. With instructions on how to do this (eg to replace the hack script we currently have and to have a constant load on the service mesh and not just try and hit it once like the hack script).

Basically something like:
./openshift-deploy.sh -n ${NAMESPACE} -c ${CONFIGMAP}

And have a couple of example configmaps that people can pick from (eg like the depth, breath, etc example we already have).

gbaufake commented:
@mwringe 
Some changes in 364e9f6:
- Included Ansible Install Script
- Readme
- Changed to configmap
